src/components/application_manager/src/resumption/resumption_data_processor_impl.cc

```diff
@@ -311,8 +311,11 @@ void ResumptionDataProcessorImpl::FinalizeResumption(
   } else {
     SDL_LOG_ERROR("Resumption for app " << app_id << " failed");
     callback(mobile_apis::Result::RESUME_FAILED, "Data resumption failed");
-    RevertRestoredData(application_manager_.application(app_id));
+    auto app = application_manager_.application(app_id);
+    RevertRestoredData(app);
     application_manager_.state_controller().DropPostponedWindows(app_id);
+    auto& builder = app->display_capabilities_builder();
+    builder.StopWaitingForWindows();
```

Comment:
can these two lines be in DeleteWindowsSubscriptions called from RevertRestoredData?

Reply:
I think yes, let me check the regression and I will push the updates

Reply:
@iCollin It seems like this issue have the same root cause as https://github.com/smartdevicelink/sdl_core/issues/3542 
We need some time for the investigation but it seems like this fix should be reworked. 
I'll keep you posted